pinot-spi/src/main/java/org/apache/pinot/spi/stream/MessageBatch.java

```diff
@@ -82,6 +88,13 @@ default StreamPartitionMsgOffset getNextStreamParitionMsgOffsetAtIndex(int index
     return new LongMsgOffset(getNextStreamMessageOffsetAtIndex(index));
   }
 
+  /**
+   * @return last offset in the batch
+   */
+  default StreamPartitionMsgOffset getLastOffset() {
+    return getNextStreamParitionMsgOffsetAtIndex(getMessageCount() - 1);
```

Comment:
Overridden for Kafka 2.0 only. I don't want to expand the scope to other implementations if it slows down fixing the Kafka bug.

Reply:
Indeed, `getMessageCount() - 1` is not an offset, but an index into the batch of messages - it returns the next offset of the last message. `getNextStreamParitionMsgOffsetAtIndex` takes an _index_ and produces the next offset from the message at that index. Does this make sense?